subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -229,12 +230,23 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
 		ter_dptr++;
 	}
 
-	/* TODO: ACAD */
-
-	/* AdvData */
-	/* Calc previous tertiary PDU len */
-	ter_len_prev = ull_adv_aux_hdr_len_calc(ter_com_hdr_prev,
-						&ter_dptr_prev);
+	/* Calc previous ACAD len and update PDU len */
+	ter_len_prev = ter_dptr_prev - (uint8_t *)ter_com_hdr_prev;
+	if (ter_len_prev < (ter_com_hdr_prev->ext_hdr_len +
```

Comment:
Multi-line conditions are hard to read.
Consider breaking the condition up into some named (const) variables. Comes at zero run-time cost.